### PR TITLE
Depl-158747 Tomcat memory and GC adjustments

### PR DESCRIPTION
**Pull request by KupchinskiyVasiliy**

### CODE REVIEW

File: setup/tomcat/conf/conf.d/setmem.conf [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    TOTAL_MEM_KB="$(awk '/MemTotal:/ { print $2 }' /proc/meminfo)"
+
+    (( METASPACE_SIZE_KB=METASPACE_SIZE_MB / 1024 ))
+    (( HEAP_KB=(TOTAL_MEM_KB - METASPACE_SIZE_KB) * USAGE_PERCENT / 100 ))
+    (( HEAP_MB=HEAP_KB / 1024 ))
+
+    echo "$HEAP_MB"
+}
+
+# Set default and maximum metaspace size
```
Comment (asmoiseenko) [inline setup/tomcat/conf/conf.d/setmem.conf:15]: Add comment:
Recommended settings for 2 OneVizion installations is 
METASPACE_SIZE_MB="384"
METASPACE_MAX_SIZE_MB="512"
Each additional installation require additional 128MB added to the each param

File: setup/tomcat/conf/conf.d/setmem.conf [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+METASPACE_SIZE_MB="384"
+METASPACE_MAX_SIZE_MB="512"
+
+CATALINA_OPTS="$CATALINA_OPTS -XX:MetaspaceSize=${METASPACE_SIZE_MB}m"
+CATALINA_OPTS="$CATALINA_OPTS -XX:MaxMetaspaceSize=${METASPACE_MAX_SIZE_MB}m"
+CATALINA_OPTS="$CATALINA_OPTS -XX:CompressedClassSpaceSize=128m"
+
+# Set maximum and minimum heap size (80% from total server memory minus metaspace size)
+HEAP_SIZE_MB="$(calculate_heapsize_mb 80 "$METASPACE_MAX_SIZE_MB")"
```
Comment (asmoiseenko) [inline setup/tomcat/conf/conf.d/setmem.conf:24]: @Simplycissimys, I think we talked about 60% of total memory for the heap. Right?

Comment (KupchinskiyVasiliy) [reply setup/tomcat/conf/conf.d/setmem.conf]: Changed to 70%, issue comments contains this number.

File: setup/tomcat/conf/conf.d/setmem.conf (hunk 1 above)
Comment (KupchinskiyVasiliy) [reply setup/tomcat/conf/conf.d/setmem.conf:15]: Done.

File: setup/tomcat/conf/conf.d/setmem.conf (hunk 2 above)
Comment (Simplycissimys) [reply setup/tomcat/conf/conf.d/setmem.conf]: 70% is ok

File: utils.sh [hunk 3]
```diff
@@ -477,6 +477,32 @@ function unpack_ps_war() {
...
 }
 
+# Uses TOMCAT_PATH variable
+function recalculate_tomcat_metaspace_size() {
+    local MEM_CONF_FILE SERVER_XML_FILE METASPACE_SIZE_MB METASPACE_MAX_SIZE_MB
+    MEM_CONF_FILE="$TOMCAT_PATH/conf/conf.d/setmem.conf"
+    SERVER_XML_FILE="$TOMCAT_PATH/conf/server.xml"
+
+    if [ ! -f "$MEM_CONF_FILE" ]; then
+        echo "Can't recalculate metaspace size! File [$MEM_CONF_FILE] is not exists"
```
Comment (asmoiseenko) [inline utils.sh:487]: change to: does not exist

File: setup/tomcat/conf/conf.d/setmem.conf (hunk 1 above)
Comment (asmoiseenko) [reply setup/tomcat/conf/conf.d/setmem.conf:15]: Sorry, my typo. Should be:
Recommended settings for 2 OneVizion installations **are**

File: utils.sh (hunk 3 above)
Comment (KupchinskiyVasiliy) [reply utils.sh]: Done.

File: setup/tomcat/conf/conf.d/setmem.conf (hunk 1 above)
Comment (KupchinskiyVasiliy) [reply setup/tomcat/conf/conf.d/setmem.conf:15]: Done.

**Pull request merged**